torch/csrc/dynamo/eval_frame.c

```diff
@@ -727,6 +727,24 @@ static int clear_state(PyObject* module) {
 
 bool is_skip_guard_eval_unsafe = false;
 
+int c_recursion_limit = -1;
+
+int get_c_recursion_limit() {
+  return c_recursion_limit;
+}
+
+static PyObject* set_c_recursion_limit_py(
+    PyObject* dummy,
+    PyObject* limit_obj) {
+  int limit = PyLong_AsInt(limit_obj);
+  if (limit == -1 && PyErr_Occurred()) {
```

Comment:
Don't you want to error out if someone tries to set a negative recursion limit? Or does it have some sort of semantic sense in CPython?
```suggestion
  if (limit < 0 || PyErr_Occurred()) {
```